src/RequestInformation.cs

```diff
@@ -199,7 +199,7 @@ private static object ReplaceEnumValueByStringRepresentation(object source)
         private static string? GetEnumName<T>(T value) where T : Enum
 #endif
         {
-            var type = value.GetType();
+            var type = typeof(T);
 
             if(Enum.GetName(type, value) is not { } name)
```

Comment:
```suggestion
            if(Enum.GetName<T>(value) is not { } name)
```

Note this API is only on `net5.0`+. So you will need a `#if`, since this still supports netstandard.

This helps in a few ways. You don't have to box `value`, and the runtime is a bit faster.

Reply:
While we are in here, you might as fix the DynamicallyAccessedMembers attribute. I think we could take one of 2 options:

1. Change `All` to be what GetMember is annotated: https://github.com/dotnet/runtime/blob/91beecdc32a893a7e2a16e4026dea7985c3aac1d/src/libraries/System.Private.CoreLib/src/System/Type.cs#L234-L240. Basically change `All` to each individual public member.
2. We know that `T` is an enum, so we really only need to call `type.GetField(name)` instead of `type.GetMember(name)`. `EnumMemberAttribute` [can only be applied to fields](https://github.com/dotnet/runtime/blob/91beecdc32a893a7e2a16e4026dea7985c3aac1d/src/libraries/System.Runtime.Serialization.Primitives/src/System/Runtime/Serialization/EnumMemberAttribute.cs#L6-L7). So we should only need to `DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)`.

Reply:
Addressed...